### PR TITLE
Update lecture 6 (MPI intro) for 2018

### PR DESCRIPTION
**Pull request by jonc125**
Mostly just tweaks needed to this lecture I think, but I've tested everything out both locally and on Aristotle to make sure I can run all the examples/exercises.

NB. I have changed the default options for reveal.js so that I can make the font size smaller on slides - some I couldn't manage to split sensibly. Hopefully this won't break any other lectures!

Main addition is a small MPI_Reduce exercise.

### CODE REVIEW

File: 06MPI/cpp/scatter.cc [hunk 1]
```diff
@@ -5,17 +5,17 @@
...
     int rank, size;
-    MPI_Comm_rank (MPI_COMM_WORLD, &rank);
-    MPI_Comm_size (MPI_COMM_WORLD, &size);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
 
     SECTION("Collective scatter") {
       /// "scatter"
       std::string const message = "This message is going to come out in separate channels";
+      if (message.size() < (unsigned)size) return;
```
Comment (klapaukh) [inline 06MPI/cpp/scatter.cc]: In this case shouldn't you just make size unsigned so you don't need to do the cast here?

File: 06MPI/sec02PointToPoint.md [hunk 2]
```diff
@@ -143,22 +147,32 @@ Why would we use Send instead of SSend?
...
+
 Both guarantee the buffer is safe to reuse. Send makes no guarantee as to
 whether it returns early or not. But *SSend* forces a *synchronisation point*:
 the codes reach the matching places, with all processes waiting until all reach
 that point.
 
 It may come out slightly faster to use Send,
-since having a **synchronisation point** when you don't need one can slow things
-down: Suppose (A) runs slightly faster, then (B) does; at the end, they've both
+since having a *synchronisation point* when you don't need one can slow things
```
Comment (klapaukh) [inline 06MPI/sec02PointToPoint.md]: Technically it is bad practice to use italics in public presentations. Compared to using bold for emphasis it causes additional problems for a large class of dyslexic readers.

Comment (jonc125) [reply 06MPI/sec02PointToPoint.md]: I didn't know that. Will fix!

File: 06MPI/cpp/scatter.cc (hunk 1 above)
Comment (jonc125) [reply 06MPI/cpp/scatter.cc]: Except the docs for MPI_Comm_size say it expects an `int`...

**Comment by jonc125** [comment]
I'm not too worried about being consistent across all lectures. I just have OCD about it when I'm writing code :)

**Comment by klapaukh** [comment]
Sure. But if you are going to be OCD about it, then surely it is better to actually make the most of it? Being consistent across all lectures seems to be a generally positive thing, and industrially would be considered good.

File: 06MPI/cpp/scatter.cc (hunk 1 above)
Comment (klapaukh) [reply 06MPI/cpp/scatter.cc]: I wonder if that is something you can file a bug with MPI for...

Comment (jonc125) [reply 06MPI/cpp/scatter.cc]: Probably not :)

File: 06MPI/sec02PointToPoint.md (hunk 2 above)
Comment (jonc125) [reply 06MPI/sec02PointToPoint.md]: Done (in lecture 6 at least). I'll do others as I see them.

**Comment by jonc125** [comment]
Yes, it would make sense. I'll create an issue for it so we don't forget.

**Pull request merged**